src/main/java/com/limechain/storage/block/BlockStateHelper.java

```diff
@@ -0,0 +1,103 @@
+package com.limechain.storage.block;
+
+import com.limechain.network.protocol.blockannounce.scale.BlockHeaderScaleWriter;
+import com.limechain.network.protocol.warp.dto.BlockHeader;
+import com.limechain.network.protocol.warp.scale.BlockHeaderReader;
+import com.limechain.storage.DBConstants;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.scale.ScaleCodecWriter;
+import io.emeraldpay.polkaj.scale.reader.UInt64Reader;
+import io.emeraldpay.polkaj.scale.writer.UInt64Writer;
+import org.javatuples.Pair;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public final class BlockStateHelper {
+    private final UInt64Writer uint64Writer = new UInt64Writer();
+    private final UInt64Reader uint64Reader = new UInt64Reader();
+
+    @NotNull
+    String headerKey(byte[] key) {
```

Comment:
Are the access modifiers missing on purpose?

Reply:
Yes, the purpose of the helper class is to unload some of the unnecessary internal methods from the BlockState class. It is marked as a final class (thus not extendable, and it's unnecessary to mark the methods as protected) and the methods are supposed to be used by the BlockState thus it is package private.